File: AppFrame11/zt-scroll-edit-sample/src/main/java/com/trade/zt_scroll_edit_sample/MyApplication.java
package com.trade.zt_scroll_edit_sample;

import android.app.Application;

/**
 * @author: zeting
 * @date: 2026/7/9
 *
 */
public class MyApplication extends Application {
    // 静态属性，通常用于存储Application实例，方便全局访问
    private static MyApplication instance;

    // 静态方法，用于获取Application实例
    public static MyApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this; // 在onCreate中初始化静态实例
    }
}

File: AppFrame11/zt_speed_device/src/main/java/com/trade/zt_speed_device/MyApplication.kt
package com.trade.zt_speed_device

import android.app.Application

/**
 * @author: zeting
 * @date: 2026/8/20
 *
 */
class MyApplication : Application() {
    // 伴生对象，用于存放静态成员（属性和方法）
    companion object {
        // 静态属性，通常用于存储Application实例，方便全局访问
        private var instance: MyApplication? = null

        // 静态方法，用于获取Application实例
        fun getInstance(): MyApplication {
            return instance as MyApplication
        }
    }

    override fun onCreate() {
        super.onCreate()
        instance = this // 在onCreate中初始化静态实例
    }
}